src/map.rs

```diff
@@ -4732,7 +4724,7 @@ where
         let reserve = if self.is_empty() {
             iter.size_hint().0
         } else {
-            (iter.size_hint().0 + 1) / 2
+            iter.size_hint().0.div_ceil(2)
```

Comment:
I'm generally not a fan of `div_ceil` because it generates [worse code](https://github.com/rust-lang/rust-clippy/issues/14944) in cases where you know the dividend is not `uX::MAX`.

Reply:
I'm fine manually allowing the lint in the repo. Although, is there a separate issue somewhere filed for the codegen issues, not just the clippy lint? Since I feel like, at least for powers of two, this should be trivial to make as good as the optimal addition + right shift.